packages/material-ui-lab/src/YearPicker/PickersYear.tsx

```diff
@@ -15,90 +39,131 @@ export interface YearProps {
   value: number;
 }
 
-export type PickersYearClassKey = 'root' | 'modeDesktop' | 'yearButton' | 'disabled' | 'selected';
+interface StyleProps extends YearProps {
+  wrapperVariant: WrapperVariant;
+}
+
+const useUtilityClasses = (styleProps: StyleProps) => {
+  const { wrapperVariant, disabled, selected, classes } = styleProps;
 
-export const styles: MuiStyles<PickersYearClassKey> = (theme): StyleRules<PickersYearClassKey> => ({
-  root: {
-    flexBasis: '33.3%',
-    display: 'flex',
-    alignItems: 'center',
-    justifyContent: 'center',
+  const slots = {
+    root: ['root', wrapperVariant && `mode${capitalize(wrapperVariant)}`],
+    yearButton: ['yearButton', disabled && 'disabled', selected && 'selected'],
+  };
+
+  return composeClasses(slots, getPickersYearUtilityClass, classes);
+};
+
+const PickersYearRoot = experimentalStyled(
+  'div',
+  {},
+  {
+    name: 'MuiPickersYear',
+    slot: 'Root',
+    overridesResolver: (
+      props: { styleProps: StyleProps },
+      styles: Record<PickersYearClassKey, object>,
+    ) => {
+      const { styleProps } = props;
+      return {
+        ...styles.root,
+        ...(styleProps.wrapperVariant &&
+          styles[`mode${capitalize(styleProps.wrapperVariant)}` as PickersYearClassKey]),
+      };
+    },
   },
-  modeDesktop: {
+)(({ styleProps }) => ({
+  flexBasis: '33.3%',
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  ...(styleProps?.wrapperVariant === 'desktop' && {
     flexBasis: '25%',
+  }),
+}));
+
+const PickersYearButton = experimentalStyled(
```

Comment:
Based on the previous name (`PrivatePickersYear`) this was a private component. Let's skip sx and remove the overrides.